apps/engmanagement/src/pages/[slug].tsx

```diff
@@ -0,0 +1,153 @@
+
+export default function ExampleChapter({post, source, authorized}: any) {
+  const {
+    title,
+    slug,
+    mainImage,
+    description,
+    ckTagId,
+    publishedAt,
+    ogImage,
+    subscribersOnly,
+  } = post
+
+  const {subscriber} = useConvertkit()
+
+  const MainImage = () => {
+    const imageProps: any = useNextSanityImage(sanityClient, mainImage)
+    if (!mainImage) return null
+    return (
+      <Image
+        src={imageProps.src}
+        {...imageProps}
+        alt=""
+        layout="fill"
+        className="md:object-contain object-cover object-top select-none pointer-events-none"
+        quality={100}
+        placeholder="blur"
+        loading="eager"
+        priority
+      />
+    )
+  }
+  return (
+    <div className="bg-[#111725]">
+      <ArticleTemplate
+        footer={authorized}
+        subscribeForm={!subscribersOnly && isEmpty(subscriber)}
+        meta={{
+          title,
+          type: 'article',
+          slug,
+          image: <MainImage />,
+          description,
+          published: publishedAt,
+          background: slug === 'the-future-of-remote-work' && (
+            <TheFutureOfRemoteWorkBackground />
+          ),
+          ogImage: ogImage && {
+            width: 1200,
+            height: 628,
+            url: ogImage.url,
+            alt: ogImage.alt,
+          },
+        }}
+      >
+        <MDXRemote {...source} />
+        {!authorized && (
+          <div className="absolute -mx-5 bottom-0 h-96 bg-gradient-to-b from-transparent to-[#111725] w-full" />
+        )}
+      </ArticleTemplate>
+      {!authorized && (
+        <section className=" max-w-screen-md w-full mx-auto sm:pb-48 pb-28">
+          <div className="flex items-start justify-center py-24 px-5 bg-gray-800 rounded-lg">
+            <div className="w-full">
+              <h3 className="sm:text-4xl text-3xl font-bold leading-none font-brandon text-center">
+                Read the rest of this article
+              </h3>
+              <h4 className="text-center text-xl text-orange-300 pt-4 pb-10">
+                This article is for subscribers only. Enter your email to
+                continue reading.
+              </h4>
+              <ConvertkitSubscribeAndTagForm tag={ckTagId} />
+            </div>
+          </div>
+        </section>
+      )}
+    </div>
+  )
+}
+
+const initialQuery = groq`*[_type == "post" && slug.current == $slug][0]{
+  title,
+  mainImage,
+  description,
+  publishedAt,
+  'slug': slug.current,
+  subscribersOnly,
+  ckTagId,
+  ogImage,
+  'body': preview
+  }`
+
+const restQuery = groq`*[_type == "post" && slug.current == $slug][0]{
+  body,
+  }`
+
+const allPostsQuery = groq`
+  *[_type == "post"]{
+    "slug": slug.current
+  }
+`
+
+export const getServerSideProps: GetServerSideProps = async (context: any) => {
+  context.res.setHeader('Cache-Control', 's-maxage=1, stale-while-revalidate')
```

Comment:
you're not going to be able to use SWR like this if the content will vary between auth'd users and non-auth'd users because the `authorized` prop will be edge cached and delivered in oddball ways between requests.

Reply:
good catch — removed, thanks!